### PR TITLE
FEATURE: Asset Constraints

### PR DESCRIPTION
**Pull request by bwaidelich**
This introduces the following Privileges:

* `ReadAssetPrivilege`
* `ReadTagPrivilege`
* `ReadAssetCollectionPrivilege`

see added documentation.

Note:
The code for this change has been taken from
https://github.com/bwaidelich/Wwwision.AssetConstraints

### CODE REVIEW

File: Neos.Media/Classes/Security/Authorization/Privilege/Doctrine/AssetAssetCollectionConditionGenerator.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        $propertyConditionGenerator = new PropertyConditionGenerator('');
+        $collectionTitle = $propertyConditionGenerator->getValueForOperand($this->collectionTitle);
+        $quotedCollectionTitle = $this->entityManager->getConnection()->quote($collectionTitle);
+
+        return $targetTableAlias . '.persistence_object_identifier IN (
+            SELECT ' . $targetTableAlias . '_a.persistence_object_identifier
+            FROM neos_media_domain_model_asset AS ' . $targetTableAlias . '_a
+            LEFT JOIN neos_media_domain_model_assetcollection_assets_join ' . $targetTableAlias . '_acj ON ' . $targetTableAlias . '_a.persistence_object_identifier = ' . $targetTableAlias . '_acj.media_asset
+            LEFT JOIN neos_media_domain_model_assetcollection ' . $targetTableAlias . '_ac ON ' . $targetTableAlias . '_ac.persistence_object_identifier = ' . $targetTableAlias . '_acj.media_assetcollection
+            WHERE ' . $targetTableAlias . '_ac.title = ' . $quotedCollectionTitle . ')';
```
Comment (aertmann) [inline Neos.Media/Classes/Security/Authorization/Privilege/Doctrine/AssetAssetCollectionConditionGenerator.php]: should be possible to use identifier as well

File: Neos.Media/Classes/Security/Authorization/Privilege/Doctrine/AssetTagConditionGenerator.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        $propertyConditionGenerator = new PropertyConditionGenerator('');
+        $tagLabel = $propertyConditionGenerator->getValueForOperand($this->tagLabel);
+        $quotedTagLabel = $this->entityManager->getConnection()->quote($tagLabel);
+
+        return $targetTableAlias . '.persistence_object_identifier IN (
+            SELECT ' . $targetTableAlias . '_a.persistence_object_identifier
+            FROM neos_media_domain_model_asset AS ' . $targetTableAlias . '_a
+            LEFT JOIN neos_media_domain_model_asset_tags_join ' . $targetTableAlias . '_atj ON ' . $targetTableAlias . '_a.persistence_object_identifier = ' . $targetTableAlias . '_atj.media_asset
+            LEFT JOIN neos_media_domain_model_tag ' . $targetTableAlias . '_t ON ' . $targetTableAlias . '_t.persistence_object_identifier = ' . $targetTableAlias . '_atj.media_tag
+            WHERE ' . $targetTableAlias . '_t.label = ' . $quotedTagLabel . ')';
```
Comment (aertmann) [inline Neos.Media/Classes/Security/Authorization/Privilege/Doctrine/AssetTagConditionGenerator.php]: should be possible to use identifier as well

**Comment by bwaidelich** [comment]
@aertmann could you check if that's what you had in mind?

**Comment by aertmann** [comment]
@bwaidelich yes exactly! thanks

will review when UUID check is fixed

**Pull request merged**